e2e/cmdenvvars/cmdenvvars.go

```diff
@@ -259,6 +259,39 @@ func (c ctx) testSingularitySypgpDir(t *testing.T) {
 	}
 }
 
+func (c ctx) testSingularityConfigDir(t *testing.T) {
+	// Test plan:
+	//
+	// - create a temporary directory to be the configuration directory
+	// - set the SINGULARITY_CONFIGDIR environment variable
+	//   to that temporary directory
+	// - run 'singularity remote list' to create the remote.yaml
+	//   file inside the configuration directory
+	// - assert that the file has been created
+	//
+	// If the file is in the temporary directory, it means
+	// singularity followed the SINGULARITY_CONFIGDIR environment
+	// variable (checked in pkg/syfs) to set the configuration directory.
+
+	configDir, cleanup := setupTemporaryDir(t, c.env.TestDir, "config-dir")
+	defer cleanup(t)
+
+	os.Setenv("SINGULARITY_CONFIGDIR", configDir)
```

Comment:
Because you set the environment variable, but do not unset it, it is causing later test to fail.

The best fix is to add something like...

```
	e2e.WithEnv(append(os.Environ(), fmt.Sprintf("SINGULARITY_CONFIGDIR=%s", configDir))
```

... to the RunSingularity call. This will ensure the environment is only set for that particular singularity invocation.